id: 5b1e53987f858500ec00
bug_link: https://syzkaller.appspot.com/bug?extid=5b1e53987f858500ec00
title: WARNING in mntput_no_expire (3)
source_page: https://syzkaller.appspot.com/upstream/fixed
trigger_commit: feb9c5e19e913b53cb536a7aa7c9f20107bb51ec
fix_commit: a91714312eb16f9ecd1f7f8b3efe1380075f28d4
datetime: '2022-05-18T02:20:17-04:00'
fix_commit_message: 'percpu_ref_init(): clean ->percpu_count_ref on failure


  That way percpu_ref_exit() is safe after failing percpu_ref_init().

  At least one user (cgroup_create()) had a double-free that way;

  there might be other similar bugs.  Easier to fix in percpu_ref_init(),

  rather than playing whack-a-mole in sloppy users...


  Usual symptoms look like a messed refcounting in one of subsystems

  that use percpu allocations (might be percpu-refcount, might be

  something else).  Having refcounts for two different objects share

  memory is Not Nice(tm)...


  Reported-by: syzbot+5b1e53987f858500ec00@syzkaller.appspotmail.com

  Signed-off-by: Al Viro <viro@zeniv.linux.org.uk>

  '
submodule:
- lib
hunk_count: 1
covered_count: 0
